core/src/main/java/org/elasticsearch/indices/recovery/RecoveryFileChunkRequest.java

```diff
@@ -105,11 +103,9 @@ public void readFrom(StreamInput in) throws IOException {
         String name = in.readString();
         position = in.readVLong();
         long length = in.readVLong();
-        String checksum = in.readOptionalString();
+        String checksum = in.readString();
         content = in.readBytesReference();
-        Version writtenBy = null;
-        String versionString = in.readOptionalString();
-        writtenBy = Lucene.parseVersionLenient(versionString, null);
+        Version writtenBy = Lucene.parseVersionLenient(in.readString(), null);
```

Comment:
Yes it has to be non-null, the only difference in using `Version.parseLeniently` being we will have to handle the `ParseException` ourselves.  I can catch the exception and wrap it into an `IOException` here